### PR TITLE
Fixed compiler warnings.

### PR DESCRIPTION
**Pull request by fjpanag**
## Summary

Fixes a couple of compiler warnings.

## Impact

N/A

## Testing

Build test.

### CODE REVIEW

File: netutils/webclient/webclient.c [hunk 1]
```diff
@@ -904,6 +906,7 @@ static inline int wget_parsechunkheader(struct webclient_context *ctx,
 static inline int wget_parsechunkenddata(struct webclient_context *ctx,
                                          struct wget_s *ws)
 {
+  UNUSED(ctx);
```
Comment (pkarashchenko) [inline netutils/webclient/webclient.c:909]: Let's move it after `ret = OK;`

File: netutils/webclient/webclient.c [hunk 2]
```diff
@@ -794,6 +795,7 @@ static inline int wget_parseheaders(struct webclient_context *ctx,
 static inline int wget_parsechunkheader(struct webclient_context *ctx,
                                         struct wget_s *ws)
 {
+  UNUSED(ctx);
```
Comment (pkarashchenko) [inline netutils/webclient/webclient.c:798]: Let's move it after `ret = OK:`

File: netutils/webclient/webclient.c [hunk 3]
```diff
@@ -1873,7 +1877,7 @@ int webclient_perform(FAR struct webclient_context *ctx)
 
               DEBUGASSERT(bytes_to_send <= ws->state_len);
               ssize_t ssz = webclient_conn_send(conn,
-                                                ws->data_buffer +
+                                                (char*)ws->data_buffer +
```
Comment (pkarashchenko) [inline netutils/webclient/webclient.c:1880]:
```suggestion
                                                (char *)ws->data_buffer +
```

File: netutils/webclient/webclient.c (hunk 1 above)
Comment (fjpanag) [reply netutils/webclient/webclient.c]: Done.

File: netutils/webclient/webclient.c (hunk 2 above)
Comment (fjpanag) [reply netutils/webclient/webclient.c]: Done.

File: netutils/webclient/webclient.c (hunk 3 above)
Comment (fjpanag) [reply netutils/webclient/webclient.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

File: netutils/webclient/webclient.c [hunk 4]
```diff
@@ -965,6 +968,7 @@ static inline int wget_parsechunktrailer(struct webclient_context *ctx,
   int offset;
   int ndx;
   int ret = OK;
+  UNUSED(ctx);
```
Comment (pkarashchenko) [inline netutils/webclient/webclient.c:971]:
```suggestion

  UNUSED(ctx);
```

File: netutils/webclient/webclient.c [hunk 5]
```diff
@@ -907,6 +909,7 @@ static inline int wget_parsechunkenddata(struct webclient_context *ctx,
...
```
Comment (pkarashchenko) [inline netutils/webclient/webclient.c:912]:
```suggestion

  UNUSED(ctx);
```

File: netutils/webclient/webclient.c [hunk 6]
```diff
@@ -797,6 +798,7 @@ static inline int wget_parsechunkheader(struct webclient_context *ctx,
...
```
Comment (pkarashchenko) [inline netutils/webclient/webclient.c:801]:
```suggestion

  UNUSED(ctx);
```

Comment (fjpanag) [reply netutils/webclient/webclient.c:802]: done

File: netutils/webclient/webclient.c (hunk 5 above)
Comment (fjpanag) [reply netutils/webclient/webclient.c:914]: done

File: netutils/webclient/webclient.c (hunk 4 above)
Comment (fjpanag) [reply netutils/webclient/webclient.c:974]: done

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
Let's ignore the know ci broken issue.

**Pull request merged**